### ISSUE TITLE
Jackson Error when entering DM channel option in slash command

### ISSUE THREAD

**Issue by Lelebees** [Discord4J/Discord4J#1171]
**To Reproduce:** 
1. Have an application / slash command that takes an option of type `7` (channel) and is available in DMs
2. Use this command in a DM, and enter the DM channel (this should be the only option)

**Expected Behavior:** 
A Channel is constructed and sent to the rest of the code

**Actual Behavior:** 
Jackson is unable to parse the incoming channel into an object.
```
Error while decoding JSON (com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `discord4j.discordjson.json.ImmutableResolvedChannelData`, problem: Cannot build ResolvedChannelData, some of required attributes are not set [name, permissions]
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: discord4j.discordjson.json.ImmutableInteractionData$Json["data"]->discord4j.discordjson.json.ImmutableApplicationCommandInteractionData$Json["resolved"]->discord4j.discordjson.json.ImmutableApplicationCommandInteractionResolvedData$Json["channels"]->java.util.LinkedHashMap["1083343032914550804"])): {"t":"INTERACTION_CREATE","s":4,"op":0,"d":{"version":1,"user":{"username":"lelebees","public_flags":4194432,"id":"373532675522166787","global_name":"Lelebees","discriminator":"0","avatar_decoration_data":null,"avatar":"2fa1d1608952a05e5b48e625e2419fbf"},"type":2,"token":"aW50ZXJhY3Rpb246MTE2Njc5NTk1OTUzMTk5NTE1NjpRSXlNNU1Fd3lpRGp4UXlTRTZSNWpQb2VUSVFHRDkyR25ySXdsRU9jOVR2ZmttUUwwWUNDUndvazZYdjF6WnlXVW0waXFGN0wwZ2xrRUpxSmk4MWlUODgySmpwZFhHd2RHVHR1ZVJtN2JoZTBOcjFNb2REYmpTQlBQSkVPbXJhQw","locale":"en-GB","id":"1166795959531995156","entitlements":[],"data":{"type":1,"resolved":{"channels":{"1083343032914550804":{"type":1,"last_message_id":"1166794957227565066","id":"1083343032914550804","flags":0}}},"options":[{"value":396118,"type":4,"name":"gameid"},{"value":"1083343032914550804","type":7,"name":"channel"}],"name":"track","id":"1166791971176718386"},"channel_id":"1083343032914550804","channel":{"type":1,"recipients":[{"username":"lelebees","public_flags":4194432,"id":"373532675522166787","global_name":"Lelebees","discriminator":"0","avatar_decoration_data":null,"avatar":"2fa1d1608952a05e5b48e625e2419fbf"}],"last_message_id":"1166794957227565066","id":"1083343032914550804","flags":0},"application_id":"887247427663450123"}}
```

**Version:** 
Discord4J 3.3.0-M2

**Other:**
I unfortunately do not have any insightful commentary.

**Comment by Gregory-Widmer** [comment]
Hello,

I successfully did reproduce the issue with the following code : 

```Java
DiscordClient.create(Optional.ofNullable(System.getenv("DISCORD_TOKEN")).or(() -> Optional.ofNullable(args.length > 0 ? args[0] : null)).orElseThrow())
        .withGateway(gatewayDiscordClient -> {

            Mono<Void> registerCommand = gatewayDiscordClient.on(ReadyEvent.class)
                    .flatMap(ignored -> {
                        ApplicationCommandRequest cmdRequest = ApplicationCommandRequest.builder()
                                .name("test")
                                .description("Test channel object")
                                .addOption(ApplicationCommandOptionData.builder()
                                        .name("channel")
                                        .description("Your channel")
                                        .type(ApplicationCommandOption.Type.CHANNEL.getValue())
                                        .required(true)
                                        .build()
                                ).build();

                        return gatewayDiscordClient.getRestClient().getApplicationId()
                                .flatMap(applicationId -> gatewayDiscordClient.getRestClient().getApplicationService()
                                        .createGlobalApplicationCommand(applicationId, cmdRequest));
                    })
                    .then();

            Mono<Void> listenForCommand = gatewayDiscordClient.on(ChatInputInteractionEvent.class)
                    // Check we got the /test command registered above
                    .filter(event -> event.getCommandName().equalsIgnoreCase("test"))
                    // Reply to the user, then do the magic :)
                    // Get the test option, if it exists and directly fetch the value by flat-mapping the optional object
                    .flatMap(event -> event.reply("Received")
                            .then(Mono.justOrEmpty(event.getOption("channel")
                                    .flatMap(ApplicationCommandInteractionOption::getValue))))
                    // Print the raw value of the option
                    .doOnNext(optValue -> System.out.printf("Found channel option with value %s\n", optValue.getRaw()))
                    // Fetch the channel object
                    .flatMap(ApplicationCommandInteractionOptionValue::asChannel)
                    .doOnNext(channel -> System.out.printf("Selected channel ID is %s, its type is %s\n", channel.getId().asString(), channel.getType().name()))
                    .then();

            return Mono.zipDelayError(
                    gatewayDiscordClient.onDisconnect(),
                    registerCommand,
                    listenForCommand
            );
        })
        .block();
```

The root cause comes from Discord provided data regarding the "resolved" channel. Because we do have a channel object, it auto-resolves it to provide up-to-date basic data regarding the channel. This is this part of the JSON data :

```JSON
{
    "data": {
      "type": 1,
      "resolved": {
        "channels": {
          "1172838873936707584": {
            "type": 1,
            "last_message_id": null,
            "id": "1172838873936707584",
            "flags": 0
          }
        }
      },
      "options": [
        {
          "value": "1172838873936707584",
          "type": 7,
          "name": "channel"
        }
      ],
      "name": "test",
      "id": "1172838574073331803"
    }
}
```

From what I can read on Discord API doc, there is a misunderstanding of the partial channel object : https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-resolved-data-structure

This did lead to make `name` and `permissions` fields mandatory where they are not in the channel object description : https://discord.com/developers/docs/resources/channel#channel-object.

This is definitively fixable but this will take some time for updating the discordjson repository and migrating the code.

I will write a PR for this.

**Comment by Gregory-Widmer** [comment]
Hello, I wrote a fix in #1176
